lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3366,6 +3366,13 @@ namespace {
         }
 
         if (isa<TypeDecl>(member)) {
+          if (auto record = dyn_cast<clang::RecordDecl>(nd)) {
+            // If this record only being declared inside of "decl" then simply
+            // ignore it because it's a forward declaration.
+            if (record->getDefinition() != record)
```

Comment:
Could you add a test for forward declaration without definition?

Are you only interested in skipping friend declarations? If so we could make the check more specific.

Will your change do the right thing (which I guess is importing the definition) for this snippet?

```cc
struct Foo {
  struct Bar;
};

struct Foo::Bar {
  int a;
};
```

Reply:
I'll keep the current test and add this one. I think it's beneficial to have both because the friend declaration isn't scoped.

Reply:
I tried adding that test but it, unfortunately, requires #34869. Once that lands, I'll update this patch with your suggested test.